internal/impl/core/metrics.go

```diff
@@ -115,98 +119,111 @@ type ccsmpBackedMetrics struct {
+	return atomic.LoadUint64(&backedMetrics.metrics[metric])
 }
 
-func (metrics *ccsmpBackedMetrics) GetStat(metric metrics.Metric) uint64 {
+func (backedMetrics *ccsmpBackedMetrics) GetStat(metric metrics.Metric) uint64 {
 	if rxMetric, ok := rxMetrics[metric]; ok {
-		return metrics.getRXStat(rxMetric)
+		// check for duplicate counts due to auto-acks and remove from final result
+		rxStat := backedMetrics.getRXStat(rxMetric)
+		duplicateAck := uint64(0)
+		// take the difference and retrun as the settled accepted metric
+		if duplicateAck > 0 && metric == metrics.PersistentMessagesAccepted {
+			return rxStat - duplicateAck
+		}
+		return rxStat // return other types of metrics as is
 	} else if txMetric, ok := txMetrics[metric]; ok {
-		return metrics.getTXStat(txMetric)
+		return backedMetrics.getTXStat(txMetric)
 	} else if clientMetric, ok := clientMetrics[metric]; ok {
-		return metrics.getNextGenStat(clientMetric)
+		return backedMetrics.getNextGenStat(clientMetric)
 	}
 	logging.Default.Warning("Could not find mapping for metric with ID " + fmt.Sprint(metric))
 	return 0
 
 }
 
-func (metrics *ccsmpBackedMetrics) ResetStats() {
+func (backedMetrics *ccsmpBackedMetrics) ResetStats() {
 	for i := 0; i < metricCount; i++ {
-		atomic.StoreUint64(&metrics.metrics[i], 0)
+		atomic.StoreUint64(&backedMetrics.metrics[i], 0)
 	}
-	metrics.resetNativeStats()
+	atomic.StoreUint64(&backedMetrics.duplicateAcks, 0) // reset this duplicate acks counter to zero
+	backedMetrics.resetNativeStats()
 }
 
-func (metrics *ccsmpBackedMetrics) resetNativeStats() {
-	metrics.metricLock.Lock()
-	defer metrics.metricLock.Unlock()
-	if metrics.capturedRxMetrics != nil {
-		for key := range metrics.capturedRxMetrics {
-			metrics.capturedRxMetrics[key] = 0
+func (backedMetrics *ccsmpBackedMetrics) resetNativeStats() {
+	backedMetrics.metricLock.Lock()
+	defer backedMetrics.metricLock.Unlock()
+	if backedMetrics.capturedRxMetrics != nil {
+		for key := range backedMetrics.capturedRxMetrics {
+			backedMetrics.capturedRxMetrics[key] = 0
 		}
-		for key := range metrics.capturedTxMetrics {
-			metrics.capturedTxMetrics[key] = 0
+		for key := range backedMetrics.capturedTxMetrics {
+			backedMetrics.capturedTxMetrics[key] = 0
 		}
 	} else {
-		errorInfo := metrics.session.SolClientSessionClearStats()
+		errorInfo := backedMetrics.session.SolClientSessionClearStats()
 		if errorInfo != nil {
 			logging.Default.Warning("Could not reset metrics: " + errorInfo.String())
 		}
 	}
 }
 
-func (metrics *ccsmpBackedMetrics) IncrementMetric(metric NextGenMetric, amount uint64) {
-	atomic.AddUint64(&metrics.metrics[metric], amount)
+func (backedMetrics *ccsmpBackedMetrics) IncrementMetric(metric NextGenMetric, amount uint64) {
+	atomic.AddUint64(&backedMetrics.metrics[metric], amount)
+}
+
+func (backedMetrics *ccsmpBackedMetrics) incrementDuplicateAckCount() {
+	atomic.AddUint64(&backedMetrics.duplicateAcks, uint64(1))
```

Comment:
@oodigie , why do we need a new function and attribute for counting duplicate acks instead of incrementing the stat using the existing method?

Reply:
@TrentDaniel, because we did not have an existing method to do this.